### ISSUE TITLE
Port the image registration demo/tutorial from https://github.com/YipengHu/label-reg

### ISSUE THREAD

**Issue by tvercaut** [NifTK/NiftyNet#212]
@YipengHu wrote a nice demo/tutorial as part of his weakly-supervised registration work: https://github.com/YipengHu/label-reg

NiftyNet already has the core components implemented but porting the demo as well would be attractive:
- [original tutorial](https://github.com/YipengHu/label-reg/blob/master/README.md)
- [label_driven_registration application in NiftyNet](https://github.com/NifTK/NiftyNet/blob/dev/niftynet/application/label_driven_registration.py)
- [interventional_dense_net in NiftyNet](https://github.com/NifTK/NiftyNet/blob/dev/niftynet/network/interventional_dense_net.py)

**Comment by YipengHu** [comment]
I seem to recall that the two-stage sampling has not been supported. @wyli Is it correct? Once this is resolved, I'd be keen to help adapt the demo/text into NiftyNet.

**Comment by wyli** [comment]
@YipengHu  that's right, training data augmentation needs some additional work as well,
otherwise we are precisely following your experiments.

- [ ] two-stage clustering sampling
- [ ] integrate random affine augmentation on the fly
